File: 43_Multiply_Strings.py
class Solution:
    def multiply(self, num1: str, num2: str) -> str:
        l1=len(num1)
        l2=len(num2)
        dicts={"1":1,"0":0,"3":3,"2":2,"4":4,"5":5,"6":6,"7":7,"8":8,"9":9}
        final=0
        f=0
        p=l1
        for j in range(l2):
            l1=p
            if num2[l2-1-j] in dicts:
                
                carry=v=k=0
                
                while l1!=0:
                    c= dicts[num1[l1-1]]*dicts[num2[l2-1-j]]*pow(10,k)
                    l1-=1
                    k+=1
                    v+=c
                    
                    
            final+=pow(10,f)*v        
            f+=1        
            
            
            
        
        return str(final)
                
        

File: 43_Multiply_Strings.cpp
class Solution {
public:
    string multiply(string nums1, string nums2) {
        if(nums1=="0" || nums2=="0")
            return "0";
        
        int carry=0;
        int length=219;
        
        int ind;
        char a[nums1.length()+nums2.length()];
        int result;
        int num;
        memset(a,'0',nums1.length()+nums2.length());
        
        int i=nums1.size()-1;
        for(;i>=0;--i)
        {
            ind=nums1.size()+nums2.length()-nums1.length()+i;
            for(int j=nums2.length()-1;j>=0;--j)
            {
                result=carry+(nums1[i]-'0')*(nums2[j]-'0');
                carry=((a[ind]-'0')+result)/10;
                a[ind]=(((a[ind]-'0')+result)%10+'0');
                ind--;
                
            }
            
            while(carry>0)
            {
                result=carry;
                
                carry=((a[ind]-'0')+result)/10;
                a[ind]=(((a[ind]-'0')+result)%10 + '0');
                ind--;
                
            }
            
            if(length>ind)
                length=ind;
        }
        
        return string(&a[length+1],&a[nums1.length()+nums2.length()]);
    }
};